### PR TITLE
State: fixup stored cards schema, stub noisy console warn on test

### PR DESCRIPTION
**Pull request by gwwar**
I missed this in testing #7697 This PR updates the stored-card schema to match the api response.

## Testing Instructions
- Clear IndexedDB
- Set debug to `localStorage.debug = 'calypso:state;'` 
- Make sure your user has a stored credit card
- Navigate to http://calypso.localhost:3000/checkout/:siteId:/personal
- Reload the page
- Check that the initial state contains `storedCard` in
screen shot 2016-08-29 at 9 41 52 am
- Check that we have no schema validation errors in console
- We can still checkout normally

cc @lamosty 

Test live: https://calypso.live/?branch=fix/stored-card-schema

### CODE REVIEW

File: client/state/stored-cards/test/reducer.js [hunk 1]
```diff
@@ -20,8 +20,13 @@ import {
 } from 'state/action-types';
 import reducer, { items } from '../reducer';
 import { STORED_CARDS_FROM_API } from './fixture';
+import { useSandbox } from 'test/helpers/use-sinon';
 
 describe( 'items', () => {
+	useSandbox( ( sandbox ) => {
+		sandbox.stub( console, 'warn' );
```
Comment (lamosty) [inline client/state/stored-cards/test/reducer.js]: Q: what is this good for?

Comment (gwwar) [inline client/state/stored-cards/test/reducer.js]: We expect a warning in one of our tests. If we didn't stub, we'd see the warning output while running the test.

eg compare the output of the following with and without the stub:
```
npm run test-client client/state/stored-cards/test/reducer.js
```

**Comment by lamosty** [comment]
LGTM and thanks :ship:.

File: client/state/stored-cards/test/reducer.js (hunk 1 above)
Comment (lamosty) [inline client/state/stored-cards/test/reducer.js]: With that stub, I don't see any warnings. Is that expected?

Comment (gwwar) [inline client/state/stored-cards/test/reducer.js]: Yup. This gets pretty noisy in the test suite. We can still spy to see if the stub has been called, but we already have pretty good coverage.

**Comment by gwwar** [comment]
Thanks for reviewing @lamosty !

**Pull request merged**